crates/workers/src/workers.rs

```diff
@@ -307,24 +327,51 @@ impl Workers {
         Ok(())
     }
 
-    /// Retrieves the worker ID associated with the specified `deal_id`.
+    /// Activates the worker with the specified `worker_id`.
+    ///
+    /// The activation process sets the worker's status to `true`, indicating that the worker
+    /// is active. The updated status is persisted, and internal data structures are updated.
     ///
     /// # Arguments
     ///
-    /// * `deal_id` - The unique identifier (`String`) of the deal associated with the worker.
+    /// * `worker_id` - The `PeerId` of the worker to be activated.
     ///
     /// # Returns
     ///
-    /// Returns `Result<PeerId, WorkersError>` where:
-    /// - `Ok(worker_id)` if the worker ID is successfully retrieved.
-    /// - `Err(WorkersError)` if an error occurs, such as the worker not found.
+    /// Returns `Result<(), WorkersError>` where:
+    /// - `Ok(())` if the activation is successful.
+    /// - `Err(WorkersError)` if an error occurs during the activation process.
     ///
-    pub fn get_worker_id(&self, deal_id: DealId) -> Result<WorkerId, WorkersError> {
-        self.worker_ids
+    pub async fn activate_worker(&self, worker_id: WorkerId) -> Result<(), WorkersError> {
+        self.set_worker_status(worker_id, true).await?;
+        Ok(())
+    }
+
+    /// Deactivates the worker with the specified `worker_id`.
+    ///
+    /// The deactivation process sets the worker's status to `false`, indicating that the worker
+    /// is not active. The updated status is persisted, and internal data structures are updated.
+    ///
+    /// # Arguments
+    ///
+    /// * `worker_id` - The `PeerId` of the worker to be deactivated.
+    ///
+    /// # Returns
+    ///
+    /// Returns `Result<(), WorkersError>` where:
+    /// - `Ok(())` if the deactivation is successful.
+    /// - `Err(WorkersError)` if an error occurs during the deactivation process.
+    ///
+    pub async fn deactivate_worker(&self, worker_id: WorkerId) -> Result<(), WorkersError> {
+        self.set_worker_status(worker_id, false).await?;
+        Ok(())
+    }
+
+    pub fn get_handle(&self, worker_id: WorkerId) -> Option<Handle> {
```

Comment:
what is a handle? File handle?

```suggestion
    pub fn get_runtime_handle(&self, worker_id: WorkerId) -> Option<Handle> {
```